### PR TITLE
Build swift inspect in swift-toolchain.yml

### PR DESCRIPTION
**Pull request by kendalharland**
Please ignore this PR: This is just for testing.  The final PR will reviewed in compnerd/swift#730 then merged into this fork, later.

### CODE REVIEW

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -2088,7 +2088,13 @@ jobs:
...
       - name: Install swift-format
         run: cmake --build ${{ github.workspace }}/BinaryCache/swift-format --target install
-
+      - name: Install Swift Toolchain
+        uses: compnerd/gha-setup-swift@8f43ccc3e8bac89829862af09de9567c807c1c12
+        with:
+          github-repo: thebrowsercompany/swift-build
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          release-asset-name: installer-amd64.exe
+          release-tag-name: '20231016.0'
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:2097]: Please don't do this - we want to use the just built Swift for building swift-inspect.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: .github/workflows/swift-toolchain.yml (hunk 1 above)
Comment (kendalharland) [reply .github/workflows/swift-toolchain.yml:2097]: What is the path to the just-built swift, and what jobs must be complete before I can run this job using that swift?

Comment (compnerd) [reply .github/workflows/swift-toolchain.yml:2097]: That depends on the pipeline configuration. See some of the other builds in this stage for the path.

This stage should have everything built - you just need the compilers and SDK built. For swift-inspect, you will also need SPM, but that was just built in this stage.

File: .github/workflows/swift-toolchain.yml [hunk 2]
```diff
@@ -2131,16 +2239,28 @@ jobs:
         arch: ['amd64' , 'arm64']
 
     steps:
+      - name: Download Debugging Tools
+        uses: actions/download-artifact@v4
+        # TODO(DEVIN-1325)
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:2244]: This is meaningless, please remove all references to internal issues unless we open up access to Linear. You should move the linear issue to GH and reference that.

File: .github/workflows/swift-toolchain.yml [hunk 3]
```diff
@@ -2120,9 +2120,117 @@ jobs:
...
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: amd64
+            cpu: x86_64
+            triple: x86_64-unknown-windows-msvc
+
+          # TODO(DEVIN-1325): Fix the arm64 build and uncomment this.
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:2138]: Please move the linear issue to GH and reference that.

**Review by compnerd — CHANGES_REQUESTED** [review]

**Review by tristanlabelle — COMMENTED** [review]
Looks good to me! Let's port to `compnerd/swift-build` for merging

**Comment by kendalharland** [comment]
Merged into compnerd/swift-build and has already been rolled into this project.

**Pull request closed without merging**